pkg/node/connection_pool.go

```diff
@@ -141,3 +141,38 @@ func (pool *ConnectionPool) Init() {
 	pool.items = make(map[storj.NodeID]*Conn)
 	pool.mu.Unlock()
 }
+
+// getAndMaintain handles all pool maintenance and returns a connection
+func (pool *ConnectionPool) getAndMaintain(node *pb.Node) *Conn {
```

Comment:
See the comment how to handle this whole thing better. If we end up disconnecting a node that is in midst of pinging then this would start causing flaky behaviour.